SetRootPW.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+# set -euo pipefail
+#
+# shellcheck disable=SC2015
+#
+# Simple script to enable maintenance-login before sealing up the template
+#
+######################################################################
+PROGNAME="$( basename "${0}" )"
+CHROOT="${CHROOT:-/mnt/ec2-root}"
+ROOTPWSTRING="${PWSTRING:-UNDEF}"
+MAINTUSER="${MAINTUSER:-root}"
+
+
+# Error handler function
+function err_exit {
+   local ERRSTR
+   local ISNUM
+   local SCRIPTEXIT
+
+   ERRSTR="${1}"
+   ISNUM='^[0-9]+$'
+   SCRIPTEXIT="${2:-1}"
+
+   if [[ ${DEBUG} == true ]]
+   then
+      # Our output channels
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+# Print out a basic usage message
+function UsageMsg {
+   local SCRIPTEXIT
+   local PART
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-h' 'Print this message'
+      printf '\t%-4s%s\n' '-m' 'Template maintenance user (default: "root")'
+      printf '\t%-4s%s\n' '-p' 'Password to assign to template maintenance user'
+      printf '\t%-6s%s\n' '' 'Default layout:'
+      for PART in ${DEFGEOMARR[*]}
+      do
+         printf '\t%-8s%s\n' '' "${PART}"
+      done
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--help' 'See "-h" short-option'
+      printf '\t%-20s%s\n' '--maintuser' 'See "-m" short-option'
+      printf '\t%-20s%s\n' '--password' 'See "-p" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+function SetPassString {
+   printf "Setting password for %s... " "${MAINTUSER}"
+   echo "${ROOTPWSTRING}" | chroot "${CHROOT}" /bin/passwd --stdin "${MAINTUSER}" && \
```

Comment:
Usually cleaner to separate the && condition, since chaining && ... || has odd failure cases.

```suggestion
   echo "${ROOTPWSTRING}" | chroot "${CHROOT}" /bin/passwd --stdin "${MAINTUSER}" || \
      err_exit "Failed setting password for ${MAINTUSER}" 1
   echo "Success"
```